libbeat/idxmgmt/ilm/error.go

```diff
@@ -0,0 +1,94 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
```

Comment:
I have seen similar functionality in `go-txfile`. It is definitely a better approach for error handling compared to what Golang provides by default. Do you have any plans to consolidate error handling? E.g moving the functions to libbeat and leaving the error variables here.

Reply:
We also have some similar functionality in go-ucfg.

> Do you have any plans to consolidate error handling? 

Yes. See: https://github.com/urso/ecslog/tree/master/errx